src/Symfony/Component/Process/Process.php

```diff
@@ -1237,14 +1237,31 @@ public static function isPtySupported()
             return $result;
         }
 
-        if ('\\' === DIRECTORY_SEPARATOR) {
+        if (self::isWindowsPlatform()) {
             return $result = false;
         }
 
         return $result = (bool) @proc_open('echo 1', array(array('pty'), array('pty'), array('pty')), $pipes);
     }
 
     /**
+     * Returns whether the current process is run on Windows platform (Command Prompt, PowerShell) or not.
+     * 
+     * Keep in mind, that Windows users can use Unix shells (like the git-shell) with TTY/PTY support,
+     * in this case, this function will return 'false'.
+     *
+     * @return bool
+     */
+    public static function isWindowsPlatform()
```

Comment:
I had it declared private in the beginning. But then I thought it would be nice to know from outside  if the process runs on `WindowsPlatform`. Like the public function `isTty`.

I could take it one step further and let the user set `windowsPlatform` (and init the value in the constructor):
```
private $windowsPlatform = false;
public function __construct() {
   if ('\\' === DIRECTORY_SEPARATOR && (!file_exists('/dev/tty') || !is_readable('/dev/tty'))) {
      $this->windowsPlatform = true;
   }
}
public function setWindowsPlatform($bool) { }
public function isWindowsPlatform() { return $this->windowsPlatform; }